src/test/resources/auditlog/endpoints/routing/configuration_valid.yml

```diff
@@ -15,7 +15,20 @@ plugins.security:
       endpoint2:
         type: external_opensearch
         config:
-          http_endpoints: ['localhost:9200','localhost:9201','localhost:9202']
+          http_endpoints: [
+            'localhost:9200',
+            'localhost:9201',
+            'localhost:9202',
+            'localhost:9202/opensearch',
+            '127.0.0.1:9200',
+            '127.0.0.1:9200/opensearch',
+            'my-opensearch-cluster.company.com:9200',
+            'my-opensearch-cluster.company.com:9200/opensearch',
+            'https://my-opensearch-cluster.company.com:9200',
```

Comment:
Is it valid to have the protocol here? For the code that extracts the url w/o port in this PR

```
int lastIndexColon = server.lastIndexOf(':');
String url = server.substring(0, lastIndexColon);
```

would that extract the value of URL as `https://my-opensearch-cluster.company.com`?

If a protocol is configurable here should it be used passed in as the scheme when the instantiating the HttpHost?

i.e.

```
if (urlContainsScheme) {
    return new HttpHost(schemeFromUrl, url, port);
} else {
    return new HttpHost(ssl ? "https" : "http", url, port);
}
```

Reply:
> Is it valid to have the protocol here? For the code that extracts the url w/o port in this PR

Yes, it works; it extracted the URL correctly and passed the test. 
 
> would that extract the value of URL as `https://my-opensearch-cluster.company.com`?

Yes, that's the result when the URL is like `https://my-opensearch-cluster.company.com:9200`.
However, if there isn't port number, it won't work. I didn't consider URLs without port number, since this is a basic configuration that users should add in order to have the full address for their OpenSearch.

@cwperks should I leave the code as it is or should I update it to make it work even when there are not port number?

Reply:
Oh, I just realized that even though all the tests passed, it didn't give me a right answer because when it's creating the hosts it adds another http or https to the URL.
I'll fix it.